src/components/Reports/SavingsFundTransfer/TransferModal/TransferModal.tsx

```diff
@@ -177,217 +275,305 @@ export const TransferModal: React.FC<TransferModalProps> = ({
+                          display: 'flex',
+                          justifyContent: 'center',
+                          alignItems: 'center',
+                        }}
+                      >
+                        <IconButton
+                          onClick={() => {
+                            setFieldValue('transferFrom', transferTo);
+                            setFieldValue('transferTo', transferFrom);
+                          }}
+                          color="primary"
+                          disabled={!transferFrom || !transferTo}
+                        >
+                          <Tooltip title={t('Swap')}>
+                            <SwapHorizIcon />
+                          </Tooltip>
+                        </IconButton>
+                      </Grid>
+
+                      <Grid item xs={12} sm={5.5}>
+                        <FormControl fullWidth>
+                          <InputLabel id="transferTo">
+                            {t('To Account')}
+                          </InputLabel>
+                          <TransferModalSelect
+                            notSelected={transferFrom}
+                            funds={funds}
+                            label={t('To Account')}
+                            labelId="transferTo"
+                            name="transferTo"
+                            value={transferTo}
+                            onChange={handleChange}
+                            onBlur={handleBlur}
+                            error={
+                              touched.transferTo && Boolean(errors.transferTo)
+                            }
+                            required
+                          />
+                          {touched.transferTo && errors.transferTo && (
+                            <FormHelperText error={true}>
+                              {errors.transferTo}
+                            </FormHelperText>
+                          )}
+                        </FormControl>
+                      </Grid>
+                    </Grid>
+                  ) : (
+                    <Grid container spacing={2} alignItems="center">
+                      <Grid
+                        item
+                        xs={12}
+                        sm={5.5}
+                        sx={{
+                          display: 'flex',
+                          justifyContent: 'flex-end',
+                          alignItems: 'center',
+                        }}
+                      >
+                        <FundInfoDisplay fund={fund as Fund} />
+                      </Grid>
+                      <Grid item xs={12} sm={1} sx={{ textAlign: 'center' }}>
+                        <East />
+                      </Grid>
+                      <Grid item xs={12} sm={5.5}>
+                        <FundInfoDisplay
+                          fund={
+                            funds.find((f) => f.fundType === transferTo) as Fund
```

Comment:
If the fund can't be found, we probably want to not render `FundInfoDisplay`.

```suggestion
                            funds.find((f) => f.fundType === transferTo)
```